api/lib/domain/services/certification-badges-service.js

```diff
@@ -78,14 +80,26 @@ function _keepHighestBadgeWithinDroitCertification(certifiableBadgeAcquisitions)
   return [...nonPixDroitBadgeAcquisitions, expertBadgeAcquisition || maitreBadgeAcquisition];
 }
 
-function _keepHighestBadgeWithinEduCertification(certifiableBadgeAcquisitions) {
-  const [pixEduBadgeAcquisitions, nonPixEduBadgeAcquisitions] = _.partition(
+function _keepHighestBadgeWithinEdu2ndDegreCertification(certifiableBadgeAcquisitions) {
+  const [pixEdu2ndDegreBadgeAcquisitions, nonPixEdu2ndDegreBadgeAcquisitions] = _.partition(
     certifiableBadgeAcquisitions,
-    (badgeAcquisition) => badgeAcquisition.isPixEdu()
+    (badgeAcquisition) => badgeAcquisition.isPixEdu2ndDegre()
   );
-  if (pixEduBadgeAcquisitions.length === 0) return nonPixEduBadgeAcquisitions;
-  const pixEduBadgeAcquisitionOrderer = new PixEduBadgeAcquisitionOrderer({
-    badgesAcquisitions: pixEduBadgeAcquisitions,
+  if (pixEdu2ndDegreBadgeAcquisitions.length === 0) return nonPixEdu2ndDegreBadgeAcquisitions;
+  const pixEduBadgeAcquisitionOrderer = new PixEdu2ndDegreBadgeAcquisitionOrderer({
+    badgesAcquisitions: pixEdu2ndDegreBadgeAcquisitions,
   });
-  return [...nonPixEduBadgeAcquisitions, pixEduBadgeAcquisitionOrderer.getHighestBadge()];
+  return [...nonPixEdu2ndDegreBadgeAcquisitions, pixEduBadgeAcquisitionOrderer.getHighestBadge()];
+}
+
+function _keepHighestBadgeWithinEdu1erDegreCertification(certifiableBadgeAcquisitions) {
+  const [pixEdu1erDegreBadgeAcquisitions, nonPixEdu1erDegreBadgeAcquisitions] = _.partition(
+    certifiableBadgeAcquisitions,
+    (badgeAcquisition) => badgeAcquisition.isPixEdu1erDegre()
+  );
+  if (pixEdu1erDegreBadgeAcquisitions.length === 0) return nonPixEdu1erDegreBadgeAcquisitions;
+  const pixEduBadgeAcquisitionOrderer = new PixEdu1erDegreBadgeAcquisitionOrderer({
+    badgesAcquisitions: pixEdu1erDegreBadgeAcquisitions,
+  });
+  return [...nonPixEdu1erDegreBadgeAcquisitions, pixEduBadgeAcquisitionOrderer.getHighestBadge()];
```

Comment:
On a 3 fois la "meme" fonction,  on pourrait factoriser: 
```suggestion
function _keepHighestBadge(certifiableBadgeAcquisitions, getterName, ordererClass) {
  const [badgesAcquisitions, nonBadgesAcquisitions] = _.partition(
    certifiableBadgeAcquisitions,
    (badgeAcquisition) => badgeAcquisition[getterName]()
  );
  if (badgesAcquisitions.length === 0) return nonBadgesAcquisitions;
  const acquisitionOrderer = new new ordererClass.prototype.constructor({
    badgesAcquisitions,
  });
  return [...nonBadgesAcquisitions, badgesAcquisitions.getHighestBadge()];
```